compiler/passes/ResolveScope.cpp

```diff
@@ -980,11 +980,11 @@ Symbol* ResolveScope::lookupPublicImports(const char* name) const {
   Symbol *retval = NULL;
 
   for_vector_allowing_0s(VisibilityStmt, visStmt, mUseImportList) {
-    if (ImportStmt *is = toImportStmt(visStmt)) {
-      if (!is->isPrivate) {
-        if (Symbol *importSym = visStmt->checkIfModuleNameMatches(name)) {
-          if (isModuleSymbol(importSym)) {
-            retval = importSym;
+    if (visStmt != NULL) {
```

Comment:
I don't know (can't remember?) how we could have NULLs in `mUseImportList`. We clearly can have and that's why we (I?) used that loop there. But the fact that there can be NULLs there seems a bit unfortunate.

Clearly, not part of this PR, thought.

Reply:
It's a way to separate the various distances for the use statements - they're traversed in a breadth-first manner, and we only want symbols at one distance to potentially conflict with symbols at the same distance, rather than conflicting with all symbols visible due to use or import statements

Reply:
Ah yeah, now I remember. Still, I don't like that we have to write this check. But don't have a quick-to-implement solution for that.

Reply:
Yeah . . . maybe like a vector of vectors, where index i corresponds to distance i?  I'll open a low priority refactor issue

Reply:
Right. May be obvious but it'd be great if that refactor adds a macro to iterate such vector of vectors in the needed order.

Reply:
I opened #15858